src/js/components/PromptDialogComponent.jsx

```diff
@@ -0,0 +1,90 @@
+var React = require("react/addons");
+var classNames = require("classnames");
+
+var Util = require("../helpers/Util");
+var ModalComponent = require("../components/ModalComponent");
+
+var PromptModalComponent = React.createClass({
+  displayName: "PromptModalComponent",
+
+  propTypes: {
+    data: React.PropTypes.shape({
+      actionButtonLabel: React.PropTypes.string.isRequired,
+      inputProperties: React.PropTypes.object.isRequired,
+      message: React.PropTypes.string.isRequired,
+      state: React.PropTypes.string.isRequired,
+      title: React.PropTypes.string.isRequired
+    }),
+    onAccept: React.PropTypes.func,
+    onDismiss: React.PropTypes.func
+  },
+
+  getDefaultProps: function () {
+    return {
+      data: null,
+      onAccept: Util.noop,
+      onDismiss: Util.noop
+    }
+  },
+
+  componentDidMount: function () {
+    let input = React.findDOMNode(this.refs.input);
+    input.focus();
+    input.select();
+  },
+
+  handleAccept: function () {
+    this.props.onAccept(React.findDOMNode(this.refs.input).value);
+  },
+
+  handleDismiss: function () {
+    this.props.onDismiss();
+  },
+
+  handleKeyUp: function (event) {
+    if (event.keyCode === 13) {
+      this.handleAccept();
+    }
+  },
+
+  render: function () {
+    var data = this.props.data;
+    var className = classNames("dialog", data.state);
+
+    return (
+      <ModalComponent
+          centered={true}
+          className={className}
+          dismissOnClickOutside={true}
+          ref="modalComponent"
+          onDestroy={this.handleDismiss}>
+        <div className="modal-header">
+          {data.title}
```

Comment:
I thought about introducing a dialog component to reduce repetition... I kept the different dialog components as we have a changing number of buttons and different contents and any component acknowledging the various different opinions  would grow into a configuration nightmare.